airflow/models.py

```diff
@@ -2023,6 +2031,21 @@ def __init__(
         self.email = email
         self.email_on_retry = email_on_retry
         self.email_on_failure = email_on_failure
+
+        self.notifiers = notifiers or {}
+        if self.email:
+            if 'email' in self.notifiers:
+                raise AirflowException("Cannot have email and notifier.email \
+                    specified at the same time")
+
+            email_on = {}
+            if self.email_on_retry:
+                email_on[State.UP_FOR_RETRY] = ['text_title.tpl', 'html_body.tpl']
+            if self.email_on_failure:
+                email_on[State.FAILED] = ['text_title.tpl', 'html_body.tpl']
+
+            self.notifiers['email'] = EmailNotifier(email, email_on)
```

Comment:
if `generaic_email_notifier = EmailNotifier(...)` was declared in module scope it could be reused by all generic tasks and decrease memory footprint of DAGs.